include/tvm/meta_schedule/integration.h

```diff
@@ -0,0 +1,214 @@
+ public:
+  /*!
+   * \brief Constructor. The name of the task extracted
+   * \brief The high-level IR
+   * \brief A list of low-level IRs that the high-level IR could potentially dispatch to
+   */
+  explicit ExtractedTask(String task_name, IRModule mod, Array<IRModule> dispatched);
+  TVM_DEFINE_NOTNULLABLE_OBJECT_REF_METHODS(ExtractedTask, runtime::ObjectRef, ExtractedTaskNode);
+};
+
+/**************** MetaScheduleContext ****************/
+
+/*!
+ * \brief A context manager interface for the integration
+ */
+class MetaScheduleContextNode : public runtime::Object {
+ public:
+  /*! \brief Default destructor */
+  virtual ~MetaScheduleContextNode() = default;
+  /*!
+   * \brief The entry point of the integration
+   * \param task_name The name of the task
+   * \param mod The high-level IR
+   * \param dispatched A list of low-level IRs that the high-level IR could potentially dispatch to.
+   * NullOpt means the dispatch needs to be done in the context.
+   * \return There are different types of the output
+   * 1) NullOpt if there is no feedback hint
+   * 2) tir::PrimFunc if `mod` should be lowered to a PrimFunc
+   * 3) relay::Function if `mod` should be dispatched to BYOC workflow
+   * 4) IRModule for unified dispatch
+   */
+  virtual Optional<ObjectRef> Query(runtime::String task_name, IRModule mod,
+                                    Optional<Array<IRModule>> dispatched) = 0;
+
+  static constexpr const char* _type_key = "meta_schedule.MetaScheduleContext";
+  TVM_DECLARE_BASE_OBJECT_INFO(MetaScheduleContextNode, runtime::Object);
+};
+
+/*!
+ * \brief Managed reference to MetaScheduleContextNode
+ * \sa MetaScheduleContextNode
+ */
+class MetaScheduleContext : public runtime::ObjectRef {
+  friend class MetaScheduleContextInternal;
+  friend class With<MetaScheduleContext>;
+
+ public:
+  /*! \brief Default destructor */
+  virtual ~MetaScheduleContext() = default;
+  /*!
+   * \brief The context manager in the current scope
+   * \return The MetaScheduleContext in the current scope. NullOpt if it's currently not under any
+   * MetaScheduleContext.
+   */
+  static Optional<MetaScheduleContext> Current();
+  /*!
+   * \brief The entry point of the integration workflow. The compilation process of the high-level
+   * IR should call this method for task extraction and for feedback hints
+   * \param task_name The name of the task
+   * \param mod The high-level IR
+   * \param dispatched A list of low-level IRs that the high-level IR could potentially dispatch to
+   * \return There are different types of the output
+   * 1) NullOpt if there is no feedback hint
+   * 2) tir::PrimFunc if `mod` should be lowered to a PrimFunc
+   * 3) relay::Function if `mod` should be dispatched to BYOC workflow
+   * 4) IRModule for unified dispatch
+   */
+  static Optional<ObjectRef> QueryInWithScope(runtime::String task_name, IRModule mod,
+                                              Optional<Array<IRModule>> dispatched);
```

Comment:
This name sounds a bit redundant. `QueryInScope` or `QueryWithScope` may be better, but this is just a nit.

Reply:
Ahhh I wanted to be as accurate as possible, it's querying inside the with-scope. Renaming to `QueryInsideWithScope` and add some code example to explain